### PR TITLE
Docs rework, prep for merge into DNET

### PR DESCRIPTION
**Pull request by csmir**
Splitting into new branch, ready for review. (Fucked up the old one, sorry :/ )

### CODE REVIEW

File: docs/faq/basics/basic-operations.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+## Why am I getting so many preemptive rate limits when I try to add more than one reactions?
+
+This is due to how HTML header works, mistreating
+0.25sec/action to 1sec. This causes the lib to throw preemptive rate
+limit more frequently than it should for methods such as adding
+reactions.
+
+## Can I opt-out of preemptive rate limits?
+
+Unfortunately, not at the moment. See [#401](https://github.com/RogueException/Discord.Net/issues/401).
```
Comment (quinchs) [inline docs/faq/basics/basic-operations.md]:
```suggestion
Unfortunately, not at the moment. See [#401](https://github.com/discord-net/Discord.Net/issues/401).
```

File: docs/faq/basics/client-basics.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+## What is a shard/sharded client, and how is it different from the `DiscordSocketClient`?
+As your bot grows in popularity, it is recommended that you should section your bot off into separate processes.
+The [DiscordShardedClient] is essentially a class that allows you to easily create and manage multiple [DiscordSocketClient]
+instances, with each one serving a different amount of guilds.
+
+There are very few differences from the [DiscordSocketClient] class, and it is very straightforward
+to modify your existing code to use a [DiscordShardedClient] when necessary.
+
+1. You can specify the total amount of shards, or shard ids, via [DiscordShardedClient]'s constructors.
```
Comment (quinchs) [inline docs/faq/basics/client-basics.md:76]:
```suggestion
1. You can specify the total amount of shards, or shard ids, via [DiscordShardedClient]'s constructors. If the total shards are not specified then the library will get the recommended shard count via the [Get Gateway Bot](https://discord.com/developers/docs/topics/gateway#get-gateway-bot) route.
```

File: docs/faq/basics/interactions.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+{
+  // pull your commands from some array, everyone has a different approach for this.
+  var commands = _builders.ToArray();
+
+  // write your list of commands globally in one go.
+  await _client.Rest.BulkOverwriteGlobalCommands(commands);
+
+  // write your array of commands to one guild in one go.
+  // You can do a foreach (... in _client.Guilds) approach to write to all guilds.
+  await _client.Rest.BulkOverwriteGuildCommands(commands, /* some guild */);
```
Comment (quinchs) [inline docs/faq/basics/interactions.md]: Should we use the methods on the `SocketGuild` class here?

File: docs/faq/commands/interaction.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+## How do I use this * interaction specific method/property?
+
+If your interaction context holds a down-casted version of the interaction object, you need to up-cast it.
+Ideally, use pattern matching to make sure its the type of interaction you are expecting it to be.
+
+## `InteractionService.ExecuteAsync()` always returns a successful result, how do i access the failed command execution results?
+
+If you are using `RunMode.Async` you need to setup your post-execution pipeline around `CommandExecuted` events.
+
+## How do i check if the executing user has * permission?
```
Comment (quinchs) [inline docs/faq/commands/interaction.md]:
```suggestion
## How do I check if the executing user has * permission?
```

File: docs/faq/commands/interaction.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+Refer to the [documentation about preconditions]
+
+## How do i send the HTTP Response from inside the command modules.
```
Comment (quinchs) [inline docs/faq/commands/interaction.md]:
```suggestion
## How do I send the HTTP Response from inside the command modules.
```

File: docs/faq/commands/interaction.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+Set the `RestResponseCallback` property of [InteractionServiceConfig] with a delegate for handling HTTP Responses and use
+[RestInteractionModuleBase] to create your command modules. `RespondAsync()` and `DeferAsync()` methods of this module base will use the
+`RestResponseCallback` to create interaction responses.
+
+## Is there a cleaner way of creating parameter choices other than using `[Choice]`?
+
+The default `enum` [TypeConverter] of the Interaction Service will
+automatically register `enum`s as multiple choice options.
+
+## How do i add an optional `enum` parameter but make the default value not visible to the user?
```
Comment (quinchs) [inline docs/faq/commands/interaction.md]:
```suggestion
## How do I add an optional `enum` parameter but make the default value not visible to the user?
```

File: docs/guides/chat_commands/intro.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
+---
+uid: Guides.ChatCommands.Intro
```
Comment (quinchs) [inline docs/guides/chat_commands/intro.md]: Should it be named "Chat Commands"? I think Text-Based Commands make more sense

**Review by quinchs — CHANGES_REQUESTED** [review]

**Comment by quinchs** [comment]
cc: @Cenngo

**Comment by quinchs** [comment]
@Rozen4334 The toc.yml needs to be mapped for the new docs

**Comment by Cenngo** [comment]
I totally forgot about this but should we at least mention that interaction service has preconditions support.

**Comment by quinchs** [comment]
> I totally forgot about this but should we at least mention that interaction service has preconditions support.

Do you want to write them up?

File: docs/faq/basics/interactions.md (hunk 3 above)
Comment (csmir) [reply docs/faq/basics/interactions.md]: @Cenngo ?

**Comment by csmir** [comment]
I am currently writing this out already, @Cenngo what do you suggest we do for dependency injection? Simplify the context but still include it?

**Comment by Cenngo** [comment]
DI patterns of Commands and Interactions are indentical apart from one detail. And i already explained that in the Resolving Module Dependencies section of intro. So its totally up to you, however you see fit.

**Comment by csmir** [comment]
Are there any types exclusive to both namespaces? I will make a glossary of types to review and look at, but will send people to the text-command docs for further detail.

**Comment by Cenngo** [comment]
Exclusive types regarding DI? No, there isnt any.

**Comment by Cenngo** [comment]
@Rozen4334 you are right, SearchSearch should be made public. I'll make a PR.

**Review by quinchs — APPROVED** [review]

**Comment by Still34** [comment]
I'll be reviewing the rest of the docs when I'm available - so far it looks like most potential fixes are mostly minor structural changes and grammatical ones. For now, it should be fine enough to read through for average users.

**Pull request merged**